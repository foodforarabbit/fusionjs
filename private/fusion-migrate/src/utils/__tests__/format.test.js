const fs = require('fs');
const path = require('path');
const execa = require('execa');
const tmp = require('tmp');
const format = require('../format.js');

test('format', async () => {
  const unformattedSource = `import a from 'b'; import c from 'd'; const thing = {
    a: 'hello'
  }`;
  const dir = tmp.dirSync().name;
  const testFile = path.join(dir, 'test.js');
  fs.writeFileSync(testFile, unformattedSource);
  expect(fs.readFileSync(testFile).toString()).toMatch(unformattedSource);
  await execa.shell('git init && git add .', {cwd: dir});
  await format(dir);
  const formattedSource = fs.readFileSync(testFile).toString();
  expect(formattedSource).toMatchSnapshot();
});
