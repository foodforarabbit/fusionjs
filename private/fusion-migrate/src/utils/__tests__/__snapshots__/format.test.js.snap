// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`format 1`] = `
"import a from 'b';
import c from 'd';
const thing = {
  a: 'hello',
};
"
`;
